### DISCUSSION TITLE
Delete group in Keycloak and have it deleted in LDAP - is it possible?

### DISCUSSION THREAD

**Discussion by oculos** [keycloak/keycloak#29000] [Q&A]
Hi,

I use `ldap_only` mode on my group mapper to LDAP. Groups are nicely sync'd, but when I delete a group in Keycloak, I'd like that group to be deleted in LDAP, since I'd like full sync between LDAP and keycloak groups. Is this possible at all, that is, to delete a group in Keycloak and have it removed from LdAP?

**Comment by raff897** [comment]
Have you enabled "Drop non-existing groups during sync " in group mapper of ldap user federation?
![Screenshot 2024-05-01 035608](https://github.com/keycloak/keycloak/assets/85362193/98e41e2e-3b55-4ced-97a8-3ead193478a3)

**Reply by oculos** [reply]
Yes. This works in this way: when I delete a group in LDAP, the group is also deleted in Keycloak. But not the opposite.

**Reply by raff897** [reply]
In keycloak 24.0 if you have enabled this feature in group mapper and the policy is LDAP_ONLY when you do sync keycloak -> federation it drop not existing group in keycloak in ldap.
Can you explain step by step your operations?

**Reply by oculos** [reply]
I have a group. The group exists both in Keycloak and in LDAP. I delete it in Keycloak, it is still in LDAP. After a few minutes, when a sync is automatically performed, the group shows up again in Keycloak.

**Reply by oculos** [reply]
@raff897 This seems to be discussed elsewhere also:

https://github.com/keycloak/keycloak/issues/29099
https://github.com/keycloak/keycloak/issues/9609

**Reply by pedroigor** [reply]
The RFE was accepted https://github.com/keycloak/keycloak/issues/29099#issuecomment-2210710506.
